packages/compartment-mapper/test/scaffold.js

```diff
@@ -87,6 +88,8 @@ export function scaffold(
 
     const application = await loadLocation(readPowers, fixture, {
       dev: true,
+      requireResolve: (from, specifier, options) =>
+        createRequire(from).resolve(specifier, options),
```

Comment:
I like how easy it is to provide the implementation.
A more secure implementation would use `options.paths` to limit the search space and guard return values by checking with `String.startsWith`that it's not reaching beyond the project folder.

A cache lookup of known needed paths for specifiers is also an option.